Servers/UI/OJS.Servers.Ui/Controllers/ContestsController.cs

```diff
@@ -89,4 +92,18 @@ public async Task<IActionResult> GetParticipatedByUser(
             .GetParticipatedByUserByFiltersAndSorting(username, model?.Map<ContestFiltersServiceModel>())
             .Map<PagedResultResponse<ContestForListingResponseModel>>()
             .ToOkResult();
+
+    /// <summary>
+    /// Gets the emails of all participants in a given contest.
+    /// </summary>
+    /// <param name="apiKey">Suls platform api key.</param>
+    /// <param name="contestId">The id of the contest.</param>
+    /// <returns>Results in json format.</returns>
+    [HttpGet]
+    [ProducesResponseType(typeof(IEnumerable<UserPercentageResultsServiceModel>), Status200OK)]
+    [Authorize(ApiKeyPolicyName)]
+    public async Task<IActionResult> GetEmailsOfParticipantsInContest(string apiKey, int? contestId)
```

Comment:
`apiKey` is redundant here. You can mark the contestId as required -  `GetEmailsOfParticipantsInContest([Required] int contestId)` and make the property non nullable, also skip the check with BusinessServiceException throwing.